raft/log.go

```diff
@@ -1021,9 +1026,11 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 			l.lock()
 			l.mustSetTermIfHigher(hb.term)
 			if hb.term >= l.term {
+				l.tracef("candidateLoop: new leader: old=%d new=%d", l.leaderID, hb.leaderID)
 				l.leaderID = hb.leaderID
 			}
 			l.unlock()
+			return Follower
```

Comment:
I don't think believe so.  From my understanding, that case block is handling heartbeats from a leader.  If it's receiving any heartbeats, then a leader has been elected and it should switch to follower state.